### PR TITLE
[5.4] Changing events property to avoid any conflicts with relations

### PR DESCRIPTION
**Pull request by arcanedev-maroc**
Fix: #17959

cc: @taylorotwell @GrahamCampbell

### CODE REVIEW

File: src/Illuminate/Database/Eloquent/Concerns/HasEvents.php [hunk 1]
```diff
@@ -152,11 +152,11 @@ protected function fireModelEvent($event, $halt = true)
      */
     protected function fireCustomModelEvent($event, $method)
     {
-        if (! isset($this->events[$event])) {
+        if (! isset($this->observableEvents[$event])) {
```
Comment (arcanedev-maroc) [inline src/Illuminate/Database/Eloquent/Concerns/HasEvents.php]: I can add a check if the Model has an array `events` property for BC.

**Comment by themsaid** [comment]
This a breaking change and should go to 5.5

**Comment by arcanedev-maroc** [comment]
Hi @themsaid, thanks for your reply.

So no need to add a check for the `event` property if exists ?

**Comment by themsaid** [comment]
I think your point is to rename the property, in that case no need to check for an `events` property imo.

**Comment by taylorotwell** [comment]
If we have the property check this can probably go in 5.4. It wouldn't be breaking in that case.

**Comment by arcanedev-maroc** [comment]
@themsaid, this is the commit (https://github.com/laravel/framework/pull/17960/commits/45e454c2add047b1f74da1b083512f91f6008506) that i included to support the old `event` property for BC.

**Comment by arcanedev-maroc** [comment]
@taylorotwell, i going to update the documentation - **Eloquent - Events section**.

No need to mention this change in the **Upgrade section**, right ?

**Comment by taylorotwell** [comment]
Hmm, on further thought I think this could have some breaking effects for packages. I would rather try to avoid this and recommend some other name for your relationship like Engagements, etc.

**Comment by arcanedev-maroc** [comment]
@taylorotwell, How about moving this change to Laravel `5.5` branch ?

**Pull request closed without merging**